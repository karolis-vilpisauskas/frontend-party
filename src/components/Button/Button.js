import React from "react";
import PropTypes from 'prop-types';
import "./Button.scss";

const Button = ({ onClick, children, styleClass, label, ...props }) => {
  return (
    <button
      {...props}
      className={`btn ${styleClass}`}
      onClick={onClick}
    >
      {children}
      <span className="btn-label">{label}</span>
    </button>
  );
};

export default Button;

Button.propTypes = {
  onClick: PropTypes.func,
  children: PropTypes.object,
  styleClass: PropTypes.string,
  label: PropTypes.string,
  props: PropTypes.array
}
