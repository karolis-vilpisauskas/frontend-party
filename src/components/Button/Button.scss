@import "styles/utils";

.btn {
  display: flex;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  transition: all 0.3s ease-in-out;
  -ms-transition-property: none !important;
  -o-transition:none;
  border: none;
  border-radius: 5px;

  &:focus {
    outline: none;
  }

  &.logout {
    color: $logout-button;
    background: transparent;
    padding: 10px;

    &:hover {
      color: $logout-hover;
      .cls-1 {
        fill: $logout-hover;
      }
    }

    .btn-label {
      margin-left: 15px;
      padding-top: 2px;
      font-size: 14px;
      font-weight: 300;
      line-height: 2.14;
      letter-spacing: 0.35px;
      text-align: left;
    }
  }

  &.login {
    max-width: 361px;
    height: 56px;
    width: 80%;

    background-color: $login-button;
    margin-top: 20px;
    color: $white;
    &:hover {
      background-color: $login-button-hover;
    }

    .btn-label {
      font-size: 16px;
      font-weight: bold;
      line-height: 1.88;
      letter-spacing: 0.4px;
    }
  }
}
