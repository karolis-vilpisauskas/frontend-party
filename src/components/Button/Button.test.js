import React from "react";
import ReactDOM from "react-dom";
import { render, cleanup, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import Button from "./Button";

afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Button />, div);
});

it("renders button correctly", () => {
  const fn = jest.fn() ;
  const { getByTestId } = render(
    <Button label="save" onClick={fn} styleClass="login">
      <span data-testid="child"></span>
    </Button>
  );
  fireEvent.click(getByTestId("btn"));
  expect(getByTestId("btn")).toHaveTextContent("save");
  expect(getByTestId("child")).toBeDefined();
  expect(getByTestId("btn").classList.contains("login")).toBe(true);
  expect(fn).toHaveBeenCalled();
});

it("matches snapshot", () => {
  const tree = renderer.create(<Button label="save"></Button>).toJSON();
  expect(tree).toMatchSnapshot();
});
