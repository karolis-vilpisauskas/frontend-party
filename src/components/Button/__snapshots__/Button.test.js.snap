// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  className="btn undefined"
  data-testid="btn"
>
  <span
    className="btn-label"
  >
    save
  </span>
</button>
`;
